packages/wrangler/src/core/register-yargs-command.ts

```diff
@@ -157,21 +159,47 @@ function createHandler(def: CommandDefinition) {
 						MIXED_MODE: args.experimentalMixedMode ?? false,
 					};
 
-			await run(experimentalFlags, () =>
-				def.handler(args, {
-					config:
-						def.behaviour?.provideConfig ?? true
-							? readConfig(args, {
-									hideWarnings: !(def.behaviour?.printConfigWarnings ?? true),
-									useRedirectIfAvailable:
-										def.behaviour?.useConfigRedirectIfAvailable,
-								})
-							: defaultWranglerConfig,
+			await run(experimentalFlags, () => {
+				const config =
+					def.behaviour?.provideConfig ?? true
+						? readConfig(args, {
+								hideWarnings: !(def.behaviour?.printConfigWarnings ?? true),
+								useRedirectIfAvailable:
+									def.behaviour?.useConfigRedirectIfAvailable,
+							})
+						: defaultWranglerConfig;
+
+				if (def.behaviour?.warnIfMultipleEnvsConfiguredButNoneSpecified) {
+					if (!args.env) {
+						const { configPath } = findWranglerConfig(process.cwd(), {
+							useRedirectIfAvailable: true,
+						});
+
+						const { rawConfig } = experimental_readRawConfig(
+							{
+								config: configPath,
+							},
+							{ hideWarnings: true }
+						);
+						const availableEnvs = Object.keys(rawConfig.env ?? {});
+						if (availableEnvs.length > 0) {
+							logger.warn(
+								dedent`
+										Multiple environments are defined in the Wrangler configuration file, but no target environment was specified for the${def.metadata.displayName ? ` ${def.metadata.displayName}` : ""} command.
```

Comment:
Can we not compute the display name of the current command?
The data is definitely in the system, but from looking at the code it is a bit awkward to pass it down to here.  Perhaps as a follow up?

Reply:
I think you have misunderstood me. My goal would be not to have to provide this field at all. Instead compute it from where the command is registered since it is already provided there.

Reply:
yeah I misunderstood your comment 😓 

Yes, I've updated this, sorry I didn't notice the `command` property before 🤦